src/chart/generateCategoricalChart.tsx

```diff
@@ -238,6 +239,21 @@ const getTooltipContent = (
       // graphic child has data props
       const entries = data === undefined ? displayedData : data;
       payload = findEntryInArray(entries, tooltipAxis.dataKey, activeLabel);
+    } else if (tooltipAxis.dataKey && tooltipAxis.allowSelectNearestValue && tooltipAxis?.categoricalDomain) {
+      const entries = data === undefined ? displayedData : data;
+      const value = tooltipAxis?.categoricalDomain?.[activeIndex];
+
+      const matchingData: any = entries.reduce((acc: any, entry: any) =>
```

Comment:
what is `acc` here? can you name that variable a bit better?

Would also love to avoid more casts and `any` where possible (I know its tough in this code, but if we don't enforce it it will keep getting worse)

Reply:
Agreed on the variable naming.

I think I would need help figuering out how to avoid ```any``` for these variables. This is a single slice of data from the data array. The data array is of type ```any[]```, to allow the user to input any sort of array of data. Please let me know if you have any thoughts on this.

Reply:
yeah this is fine until Recharts implements generics

Reply:
To be on the safe side then, I would suggest to replace `any` with `unknown`. It would cause all sorts of checks to be required, but we would be on the safe side.